### PR TITLE
Feature/account linking

### PR DESCRIPTION
**Pull request by everyplace**
This PR adds Account Linking functionality to the `reader-revenue-demo` app. This PR is deployed to a testing instance at [account-linking-upgrade](https://account-linking-upgrade-dot-reader-revenue-demo.ue.r.appspot.com/account-linking/client-side).

Significant changes:

- Adds a detailed article at `app/content/account-linking.md`
- Adds the `/api/account-linking` endpoint via the `app/routes/account-linking/api.js` router and it's helper class in `app/routes/account-linking/entitlements.js`
- Adds a client-side javascript example of triggering account linking at `public/js/account-linking.js`, and a helper class for managing a localStorage cache of user state via `public/js/account-linking-persistence.js`

### CODE REVIEW

File: public/js/account-linking-persistence.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+class accountLinkingPersistence {
+  constructor() {
+    this._linked = false;
+    this._declined = false;
+    this.accessToken = undefined;
+    this.refresh();
+    this.generateAccessToken();
+  }
+
+  set linked(item) {
```
[the remark this answers is not in the document]
Comment (ed-bird) [reply public/js/account-linking-persistence.js:30]: Change the variable name to make it clearer what `item` is

[the remark this answers is not in the document]
Comment (ed-bird) [reply public/js/account-linking-persistence.js:30]: Or could use `linkState` to align with `declined`

File: public/js/account-linking-persistence.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+/**
+ * @fileoverview Class for use with managing 
+ */
+
+class accountLinkingPersistence {
+  constructor() {
+    this._linked = false;
+    this._declined = false;
+    this.accessToken = undefined;
+    this.refresh();
```
Comment (ed-bird) [inline public/js/account-linking-persistence.js:26]: Why is this needed?

File: public/js/account-linking-persistence.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Class for use with managing 
+ */
+
+class accountLinkingPersistence {
+  constructor() {
```
Comment (ed-bird) [inline public/js/account-linking-persistence.js:22]: Real applications could check database here to see if the user has already linked / declined, add a comment to call that out?

File: public/js/account-linking-persistence.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Class for use with managing
```
Comment (ed-bird) [inline public/js/account-linking-persistence.js:18]: Finish file comment :)

**Review by ed-bird — APPROVED** [review]

File: app/content/account-linking.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+Entitlements API and other required endpoints for OAuth 2.0 flow need to be defined in Publisher Center. For more information setting up endpoints, please read Set up Subscribe with Google for your publication.
+
+The Console DevTool is available to test account linking. For more information, please read Test account linking.
+
+## Sample Implementation
+
+1. Include [swg.js](https://developers.google.com/news/reader-revenue/monetization/prerequisites/configure-javascript): `<script async src="https://news.google.com/swg/js/v1/swg.js"></script>`.
+1. Include the appropriate [structured data markup](https://developers.google.com/news/reader-revenue/monetization/prerequisites/structured-data-markup) for the page.
+1. Include and configure [Sign in with Google](https://developers.google.com/identity/gsi/web/guides/overview): `<script async src="https://apis.google.com/js/platform.js"></script>`.
```
[the remark this answers is not in the document]
Comment (everyplace) [reply app/content/account-linking.md:41]: You know what's funny is that not only are you right, but my working example doesn't include SiwG!

File: app/routes/account-linking/api.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+import express from 'express';
+import { Entitlements } from './entitlements.js';
+
+const router = express.Router();
+
+router.get('/', async (req, res)=>{
+
+  const entitlements = new Entitlements(req.query["access_token"]);
+
+  return res.json(entitlements.fetch('basic'));
```
[the remark this answers is not in the document]
Comment (everyplace) [reply app/routes/account-linking/api.js:26]: Great callout. Added.

File: public/js/account-linking-persistence.js (hunk 2 above)
Comment (everyplace) [reply public/js/account-linking-persistence.js:26]: I use `this.refresh()` in the constructor to attempt to pull down the latest state from localStorage. Theoretically the class could just do this as part of the getters, but I felt like having a record of state at instantiation would make more sense in this example, rather than querying the "store" each time a request came in.

Comment (ed-bird) [reply public/js/account-linking-persistence.js:38]: Got it, makes sense!

File: public/js/account-linking-persistence.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ed-bird) [reply public/js/account-linking-persistence.js]: Awesome, thanks!

**Review by ed-bird — APPROVED** [review]
